datashape/coretypes.py

```diff
@@ -402,6 +402,76 @@ def to_numpy_dtype(self):
         return np.dtype('O', metadata={'vlen': unicode})
 
 
+class Decimal(Unit):
+    """Decimal type corresponding to IEEE 754-2008 decimal floating point types
+    and SQL Decimal/Numeric types.
+
+    The first parameter passed specifies the number of digits of precision that
+    the Decimal contains. If an additional parameter is given, it represents
+    the scale, or number of digits of precision that are after the decimal
+    point.
+
+    The Decimal type makes no requirement of how it is to be stored in memory,
+    therefore, the number of bytes needed to store a Decimal for a given
+    precision will vary based on the platform where it is used. For example,
+    the IEEE 754-2008 standard states that a decimal32 can only contain up to
+    seven decimal digits of precision, therefore a Decimal with nine digits of
+    precision would require a decimal64, however, the Teradata documentation
+    allows for up to nine digits in a four byte Decimal, therefore Teradata
+    would effectively store a Decimal(9) as a decimal32.
```

Comment:
IEEE 754-2008 describes a floating point decimal type, whereas SQL decimal is a fixed point type. This description is written as if they're kind of the same. I suspect what you want to do is just specify the fixed point decimal as used by SQL, and leave the decimal32/64/128 as it was, because those are quite different. This comment should say how it is just for the SQL fixed point decimals, and is different from IEEE decimal.

For datashape, not specifying the storage makes sense to me. For DyND, I imagine we would want to make the same choice Impala does.

http://www.cloudera.com/content/cloudera/en/documentation/cloudera-impala/latest/topics/impala_decimal.html
```
Parquet and Avro tables use binary formats, In these tables, Impala stores each value in as few bytes as possible depending on the precision specified for the DECIMAL column.
In memory, DECIMAL values with precision of 9 or less are stored in 4 bytes.
In memory, DECIMAL values with precision of 10 through 18 are stored in 8 bytes.
In memory, DECIMAL values with precision greater than 18 are stored in 16 bytes.
```